diff --git a/React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm b/React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm
--- a/React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm
+++ b/React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm
@@ -701,7 +701,8 @@
 - (TextInputEventEmitter::Metrics)_textInputMetrics
 {
   return {
       .text = RCTStringFromNSString(_backedTextInputView.attributedText.string),
+      .isComposing = _backedTextInputView.markedTextRange != nil,
       .selectionRange = [self _selectionRange],
       .eventCount = static_cast<int>(_mostRecentEventCount),
       .contentOffset = RCTPointFromCGPoint(_backedTextInputView.contentOffset),
diff --git a/ReactCommon/react/renderer/components/textinput/TextInputEventEmitter.cpp b/ReactCommon/react/renderer/components/textinput/TextInputEventEmitter.cpp
--- a/ReactCommon/react/renderer/components/textinput/TextInputEventEmitter.cpp
+++ b/ReactCommon/react/renderer/components/textinput/TextInputEventEmitter.cpp
@@ -21,1 +21,5 @@ static jsi::Value textInputMetricsPayload(
       jsi::String::createFromUtf8(runtime, textInputMetrics.text));
+  if (textInputMetrics.isComposing.has_value()) {
+    payload.setProperty(runtime, "isComposing", *textInputMetrics.isComposing);
+  }
+
diff --git a/ReactCommon/react/renderer/components/textinput/TextInputEventEmitter.h b/ReactCommon/react/renderer/components/textinput/TextInputEventEmitter.h
--- a/ReactCommon/react/renderer/components/textinput/TextInputEventEmitter.h
+++ b/ReactCommon/react/renderer/components/textinput/TextInputEventEmitter.h
@@ -10,1 +10,3 @@
+#include <optional>
+
 #include <react/renderer/attributedstring/AttributedString.h>
@@ -20,2 +22,3 @@ class TextInputEventEmitter : public ViewEventEmitter {
   struct Metrics {
     std::string text;
+    std::optional<bool> isComposing;
